drivers/adc/adm1177/iio_adm1177.c

```diff
@@ -0,0 +1,266 @@
+*   @file   iio_adm1177.c
+*   @brief  Source file of iio_adm1177
+*   @author Radu Sabau (radu.sabau@analog.com)
+*******************************************************************************
+* Copyright 2023(c) Analog Devices, Inc.
+*
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are met:
+*  - Redistributions of source code must retain the above copyright
+*    notice, this list of conditions and the following disclaimer.
+*  - Redistributions in binary form must reproduce the above copyright
+*    notice, this list of conditions and the following disclaimer in
+*    the documentation and/or other materials provided with the
+*    distribution.
+*  - Neither the name of Analog Devices, Inc. nor the names of its
+*    contributors may be used to endorse or promote products derived
+*    from this software without specific prior written permission.
+*  - The use of this software may or may not infringe the patent rights
+*    of one or more patent holders.  This license does not release you
+*    from the requirement that you obtain separate licenses from these
+*    patent holders to use this software.
+*  - Use of the software either in source or binary form, must be run
+*    on or directly connected to an Analog Devices Inc. component.
+*
+* THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+* IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+* MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+* IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+* LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+* SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+* CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************************************************************/
+#include "iio_adm1177.h"
+#include "adm1177.h"
+#include "no_os_error.h"
+#include "no_os_util.h"
+#include "no_os_alloc.h"
+
+static struct scan_type adm1177_iio_scan_type = {
+	.sign = 'u',
+	.realbits = 12,
+	.storagebits = 32,
+	.shift = 0,
+	.is_big_endian = false,
+};
+
+static int adm1177_iio_read_raw(void *device, char *buf, uint32_t len,
+				const struct iio_ch_info *channel,
+				intptr_t priv)
+{
+	struct adm1177_iio_dev *iiodev = device;
+	struct adm1177_dev *desc = (struct adm1177_dev *)iiodev->adm1177_dev;
+	int32_t ret, value;
+	uint16_t conv_voltage;
+	uint16_t conv_current;
+
+	switch (channel->type) {
+	case IIO_VOLTAGE:
+		ret = adm1177_read(desc, &conv_voltage, &conv_current);
```

Comment:
now this function doesn't exist anymore. please replace all `adm1177_read` with the new function header.